src/core_modules/capture-core/components/WidgetsChangelog/common/hooks/useChangelogData.js

```diff
@@ -0,0 +1,118 @@
+// @flow
+import moment from 'moment';
+import { v4 as uuid } from 'uuid';
+import { useMemo, useState } from 'react';
+import { useTimeZoneConversion } from '@dhis2/app-runtime';
+import { useApiDataQuery } from '../../../../utils/reactQueryHelpers';
+import { CHANGELOG_ENTITY_TYPES, QUERY_KEYS_BY_ENTITY_TYPE } from '../../Changelog/Changelog.constants';
+import type { Change, ChangelogRecord, ItemDefinitions, SortDirection } from '../../Changelog/Changelog.types';
+import { convertServerToClient } from '../../../../converters';
+import { convert } from '../../../../converters/clientToList';
```

Comment:
As I understand it from the tech design, we want `WidgetsChangelog` to be self-contained. This means that the widget will have to implement its own converters and not import the common ones. WDYT?

Reply:
This is ok. See other comment :)